azerothcore/azerothcore-wotlk
# fix(Core): Suppress status opcode after gossip menu

This issue seems to just affect only NPCs who are quest givers who are also vendors/trainers. 

## Changes Proposed:
-  Suppress opcode query for gossip of quests


## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/2604
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/9745

## SOURCE:

This is debug while walking to NPC and right clicking
![gossip](https://user-images.githubusercontent.com/22668627/147861575-442c3412-615c-4487-b350-e0bffdce11f6.gif)


## Tests Performed:

- Built and tested on my local environment, no issues


## How to Test the Changes:


1. Checkout the master
2. Goto any vendor/trainer that is also a quest giver, (does not need to have any active quests)
3. Walk towards the npc while spam right clicking (don't need to go crazy here, just typical game playing)
4. You will now notice the menu appears from the NPC, however clicking on any menu item does nothing

Repeat test with standing still
You will notice it works fine.

Repeat walking test with this PR, you will see issue is gone

## Known Issues and TODO List:


- More testing, interacting with other NPCs


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
